File: app/src/main/java/com/tanxin/base/modules/ArouteActivity.java
package com.tanxin.base.modules;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;

import com.alibaba.android.arouter.facade.annotation.Autowired;
import com.alibaba.android.arouter.facade.annotation.Route;
import com.blankj.utilcode.util.ToastUtils;
import com.tanxin.base.R;
import com.tanxin.base.common.base.BaseActivity;
import com.tanxin.base.common.base.BaseEvent;

import butterknife.BindView;

@Route(path = "/test/activity")
public class ArouteActivity extends BaseActivity {
    @Autowired
    public String key;

    @BindView(R.id.tv)
    TextView tv;


    @Override
    protected int bindLayout() {
        return R.layout.activity_aroute;
    }

    @Override
    protected void onEventComing(BaseEvent event) {

    }

    @Override
    protected void initView(Bundle savedInstanceState) {
        ToastUtils.showShort(key);
        tv.setText(key);
    }
}

File: app/src/main/java/com/tanxin/base/modules/activity/ArouteActivity.kt
package com.tanxin.base.modules.activity

import android.content.Context
import android.content.Intent
import android.os.Bundle
import com.blankj.utilcode.util.ToastUtils
import com.tanxin.base.R
import com.tanxin.base.common.base.BaseEvent
import com.tanxin.base.modules.CeshiActivity
import com.tanxin.base.common.base.BaseActivity
import kotlinx.android.synthetic.main.activity_aroute.*
import org.jetbrains.anko.startActivity
import org.jetbrains.anko.toast


class ArouteActivity : BaseActivity() {
    var key:String? = null

    companion object {
        fun startActivity(context:Context,key: String) {
            context.startActivity(Intent(context, ArouteActivity::class.java).putExtra("key", key))
        }
    }

    override fun bindLayout(): Int = R.layout.activity_aroute

    override fun getBundleExtras(extras: Bundle) {
        key = extras.getString("key")
    }

    override fun initView(savedInstanceState: Bundle?) {
        ToastUtils.showShort(key)
        tv.text = key
        startActivity<CeshiActivity>()
    }

}
